test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/PackageSearchRunnerTests.cs

```diff
@@ -0,0 +1,389 @@
+                    "| ",
+                    "",
+                    "Fake.Newtonsoft.",
+                    "Json",
+                    "",
+                    " ",
+                    "| 12.0.3         ",
+                    "| James Newton-King ",
+                    "| 531,607,259 ",
+                };
+
+            PackageSearchArgs packageSearchArgs = new()
+            {
+                Skip = 0,
+                Take = 20,
+                Prerelease = false,
+                ExactMatch = false,
+                Logger = _fixture.GetLogger(),
+                SearchTerm = "json",
+                Sources = new List<string> { $"{mockServer.Uri}v3/index.json" }
+            };
+
+            string index = $@"
+                {{
+                    ""version"": ""3.0.0"",
+
+                    ""resources"": [
+                    {{
+                        ""@id"": ""{mockServer.Uri + "search/query"}"",
+                        ""@type"": ""SearchQueryService/Versioned"",
+                        ""comment"": ""Query endpoint of NuGet Search service (primary)""
+                    }}
+                    ],
+
+                    ""@context"":
+                    {{
+                        ""@vocab"": ""http://schema.nuget.org/services#"",
+                        ""comment"": ""http://www.w3.org/2000/01/rdf-schema#comment""
+                    }}
+                }}";
+
+            mockServer.Get.Add("/v3/index.json", r => index);
+            mockServer.Get.Add("/search/query?q=json&skip=0&take=20&prerelease=false&semVerLevel=2.0.0", r => _onePackageQueryResult);
+            mockServer.Start();
+
+            // Act
+            await PackageSearchRunner.RunAsync(
+                sourceProvider: sourceProvider,
+                packageSearchArgs,
+                cancellationToken: System.Threading.CancellationToken.None);
+
+            //stop mock server
+            mockServer.Stop();
+
+            // Assert
+            foreach (var expected in expectedValues)
+            {
+                Assert.Contains(expected, _fixture.ColoredMessage.Select(tuple => tuple.Item1));
+            }
+
+            Assert.Contains(Tuple.Create("Json", ConsoleColor.Red), _fixture.ColoredMessage);
+        }
+
+        [Theory]
+        [InlineData(0, 10, true)]
+        [InlineData(0, 20, false)]
+        [InlineData(5, 10, true)]
+        [InlineData(10, 20, false)]
+        public async Task PackageSearchRunner_SearchAPIWithVariousSkipTakePrereleaseOptionsValuesReturnsOnePackage_OnePackageTableOutputted(int skip, int take, bool prerelease)
+        {
+            // Arrange
+            ISettings settings = Settings.LoadDefaultSettings(Directory.GetCurrentDirectory(),
+                    configFileName: null,
+                    machineWideSettings: new XPlatMachineWideSetting());
+            PackageSourceProvider sourceProvider = new PackageSourceProvider(settings);
+            var mockServer = new MockServer();
```

Comment:
Anytime a mock server gets created, we need to ensure it's disposed.

```suggestion
            using (var mockServer = new MockServer())
            {
```
Or if you don't care that it gets disposed at the very end of the test method:

```suggestion
            using var mockServer = new MockServer();
```